### DISCUSSION TITLE
How to start with the main window maximized?

### DISCUSSION THREAD

**Discussion by oswald2** [slint-ui/slint#3478] [Q&A] [answered]
Is there an easy way to start with the main window maximized (desktop applications)? 

Currently, it seems only possible to go via winit -> monitor (primary display) -> video mode and get the sizes and then manually set them manually. Maybe I am overlooking something?

**Comment by ogoffart** [comment, marked as answer]
Right now, the situation is not ideal. 
You can use the `SLINT_FULLSCREEN=1`  environment variable (it is documented [here](https://slint.dev/releases/1.2.0/docs/slint/src/advanced/backend_winit) )
The other option is to use private API via the [i-slint-backend-winit](https://docs.rs/i-slint-backend-winit/latest/i_slint_backend_winit/) and then you can have a reference to the winit::Window so you can put it full screen through this API

Does that answer your question?

**Reply by oswald2** [reply]
Yes, thanks a lot for the quick answer!

**Reply by neneodonkor** [reply]
How do I disable the maximize button so the user can't maximize the window?

**Comment by syf20020816** [comment]
The following is a kind of solve :
1. get screen resolution (use rust|c++) and count height , width
2. use in property <{h:length,w:length}> screenLength;
3. main.rs -> let window = MainWindow::new().unwrap(); // pass the screen h and w
4. set Window's property no-frame : true

**Reply by neneodonkor** [reply]
Do you have an idea as to how I can disable the maximize button?

**Reply by syf20020816** [reply]
I think you can do some config when package the app. I seem to have seen these configurations before when I use cargo-packager.
seen [doc](https://docs.crabnebula.dev/packager/)
